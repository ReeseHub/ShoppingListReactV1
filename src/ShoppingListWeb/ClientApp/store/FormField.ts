export interface FormFieldState {
    className: string,
    doValidate: boolean,
    meta: any,
    theme: string,
    label: any,
    componentClass: string,
    type: string,
    placeholder: string,
    input :any
}