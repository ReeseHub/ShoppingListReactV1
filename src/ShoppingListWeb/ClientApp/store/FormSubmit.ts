export interface FormSubmitState {
    error: string,  // redux-form general `_error` message
    invalid: boolean,  // redux-form invalid prop
    submitting: boolean,   // redux-form invalid submitting
    buttonSaveLoading: string, // save button loading text, default is "Saving..."
    buttonSave: string
}