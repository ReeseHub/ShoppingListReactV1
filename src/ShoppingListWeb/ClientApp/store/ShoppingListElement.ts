export interface ShoppingListElementState {
    item: ShoppingListItem,
    showDelete: (item: ShoppingListItem)=>void,
}

export interface ShoppingListItem {

    id: number,
    itemName: string,
    quantity: number,
    category: { id: number, name: string }
}