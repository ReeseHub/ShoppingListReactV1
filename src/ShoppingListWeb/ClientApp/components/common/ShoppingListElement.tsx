import * as React from "react";
import { Link } from "react-router-dom";
import { Button, Glyphicon } from "react-bootstrap";
import * as ShoppingListElementState from '../../store/ShoppingListElement';


type ShoppingListElementState = ShoppingListElementState.ShoppingListElementState;

// Pet List Element component
export default class ShoppingListElement extends React.Component<ShoppingListElementState, {}> {
    // render
    render() {
        const { item, showDelete } = this.props;
        return (
            <tr>
                <td>#{item.id}</td>
                <td>{item.itemName}</td>
                <td>{item.quantity}</td>
                <td>{item.category.name}</td>
                <td>
                    <Link to={'shoppinglist-edit/' + item.id}>
                        <Button bsSize="xsmall">
                            Edit <Glyphicon glyph="edit" />
                        </Button>
                    </Link>
                </td>
                <td>
                    <Button bsSize="xsmall" className="pet-delete" onClick={() => showDelete(item)}>
                        Delete <Glyphicon glyph="remove-circle" />
                    </Button>
                </td>
            </tr>
        );
    }
}






