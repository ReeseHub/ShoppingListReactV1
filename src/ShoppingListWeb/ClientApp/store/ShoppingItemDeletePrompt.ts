import * as namespace from './ShoppingListElement'

export interface ShoppingItemDeletePromptState {
    show: boolean,
    item: namespace.ShoppingListItem,
    hideDelete: () => void,
    itemDelete: () => void
}