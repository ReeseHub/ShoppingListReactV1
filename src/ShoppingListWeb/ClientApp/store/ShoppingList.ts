
import * as namespace from './ShoppingListElement'

export interface ShoppingListState {
    shoppingItems: namespace.ShoppingListItem[],
    page : number
}