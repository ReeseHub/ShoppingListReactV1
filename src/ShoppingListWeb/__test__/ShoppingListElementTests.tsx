import * as React from "react";
import { HTMLAttributes, shallow, ShallowWrapper } from "enzyme";
import ShoppingListElement from "../ClientApp/components/common/ShoppingListElement";
import { ShoppingListElementState, ShoppingListItem } from "../ClientApp/store/ShoppingListElement";

import * as enzyme from 'enzyme';
import * as Adapter  from 'enzyme-adapter-react-15';

enzyme.configure({ adapter: new Adapter() });

const testShoppingListElementProps: ShoppingListElementState = {

    item: {
        id: 1,
        item: "Test",
        quantity: 2,
        category: { id: 1, name: "TestCategory" }

    },
    showDelete: (item: ShoppingListItem): void => { }


};

let child: ShallowWrapper<undefined, undefined>;



beforeEach(() =>
    child = shallow(<ShoppingListElement {...testShoppingListElementProps} />));

 //checking that all is fine and component has been rendered
it("should render without error", () => {
    console.log(child);
    expect(child.length).toBe(1));
};



test('adds 1 + 2 to equal 3', () => {

    let temp: number = 1 + 2;

    expect(temp).toBe(3);
});

    