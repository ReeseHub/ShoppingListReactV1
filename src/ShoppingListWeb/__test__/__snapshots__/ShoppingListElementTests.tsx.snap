// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td>
    #
    1
  </td>
  <td>
    Test
  </td>
  <td>
    2
  </td>
  <td>
    TestCategory
  </td>
  <td>
    <Link
      replace={false}
      to="shoppinglist-edit/1"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsSize="xsmall"
        bsStyle="default"
        disabled={false}
      >
        Edit 
        <Glyphicon
          bsClass="glyphicon"
          glyph="edit"
        />
      </Button>
    </Link>
  </td>
  <td>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="xsmall"
      bsStyle="default"
      className="pet-delete"
      disabled={false}
      onClick={[Function]}
    >
      Delete 
      <Glyphicon
        bsClass="glyphicon"
        glyph="remove-circle"
      />
    </Button>
  </td>
</tr>
`;
